packages/sit-onyx/src/composables/useTimer.ts

```diff
@@ -0,0 +1,68 @@
+import { computed, ref, watch, type Ref } from "vue";
+
+type TimerOptions = {
+  endTime: Ref<string> /** expect ISO string @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString} */;
+  isPaused?: Ref<boolean>;
+  startImmediately?: boolean;
+};
+
+/**
+ * Timer composable
+ * @param options @see TimerOptions
+ * @returns composable to start and stop a timer, get the time left and track if the timer has ended.
+ */
+export const useTimer = (options: TimerOptions) => {
+  const intervalId = ref<ReturnType<typeof setInterval>>();
+  const timeLeft = ref(0);
+  const isEnded = ref(false);
+
+  const validateEndTime = () => {
+    const date = new Date(options.endTime.value);
+    if (isNaN(date.getTime())) {
+      throw new Error("Invalid end time");
+    }
+  };
+
+  const endTimestamp = computed(() => {
+    return new Date(options.endTime.value).getTime();
+  });
+
+  const calculateTimeLeft = () => {
+    timeLeft.value = endTimestamp.value - Date.now();
+    if (timeLeft.value <= 0) {
+      timeLeft.value = 0;
+    }
+  };
+
+  const startTimer = () => {
+    clearInterval(intervalId.value);
+    intervalId.value = setInterval(() => {
+      if (endTimestamp.value > 0) {
+        if (options.isPaused?.value) return;
+        calculateTimeLeft();
+        if (timeLeft.value === 0) {
+          isEnded.value = true;
+          clearInterval(intervalId.value);
```

Comment:
WE should also clear the timer on `onBeforeUnmount()` to prevent any dangling timers